modules/hal_bouffalolab/CMakeLists.txt

```diff
@@ -0,0 +1,60 @@
+# Copyright (c) 2021-2025 Gerson Fernando Budke <nandojve@gmail.com>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+if(CONFIG_SOC_FAMILY_BOUFFALOLAB_BFLB)
+  zephyr_library_named(hal_bouffalolab)
+
+  # This is necessary to proper select architecture in the SDK
+  zephyr_compile_definitions(
+    ARCH_RISCV
+  )
+
+  zephyr_library_compile_definitions(
+    BFLB_USE_HAL_DRIVER
+    BFLB_USE_CUSTOM_LD_SECTIONS
+  )
+
+  set(bflb_soc         bl602)
```

Comment:
just an observation, but this decouples the passed to the hal from the SoC described by the soc.yml and in Kconfig, which in future can make it harder to find / handle if support for other bouffalo SoCs is introduced.

Not a request to change now, but please consider if this could be improved if addressing other comments.

Reply:
Noted. It will be improved.